### PR TITLE
[CDF-20588] MyPy Part 2

### PR DESCRIPTION
**Pull request by doctrino**
## Description

Adding `mypy` check to the `cognite_toolkit/cdf_tk/bootstrap.py`. I have done a minimal amount of changes, and instead tried to just add type hints to reflect the actual types that are being used. 

## Checklist:
- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped. [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/cdf_tk/bootstrap.py [hunk 1]
```diff
@@ -32,34 +33,34 @@
...
-    cdf_url: str = None
-    token_url: str = None
-    tenant_id: str = None
-    audience: str = None
-    scopes: str = None
+    cluster: str | None
+    project: str | None
+    token: str | None
+    client_id: str | None
+    client_secret: str | None
```
Comment (doctrino) [inline cognite_toolkit/cdf_tk/bootstrap.py:40]: I set these to `None` to reflect the behavior as it is today. The other option, is to raise an exception in line 54-58 if any of these are missing from the environment.

Comment (gregertw) [reply cognite_toolkit/cdf_tk/bootstrap.py:40]: Let's go for None for now. If we raise an exception, we might want to use the typer support for prompting if a command line option is not set either from env var or explicitly.

File: cognite_toolkit/cdf_tk/bootstrap.py [hunk 2]
```diff
@@ -32,34 +33,34 @@
...
     vars = AuthVariables(
-        cluster=os.environ.get("CDF_CLUSTER", None),
-        project=os.environ.get("CDF_PROJECT", None),
-        token=os.environ.get("CDF_TOKEN", None),
-        client_id=os.environ.get("IDP_CLIENT_ID", None),
-        client_secret=os.environ.get("IDP_CLIENT_SECRET", None),
-        cdf_url=os.environ.get("CDF_URL", None),
-        tenant_id=os.environ.get("IDP_TENANT_ID", None),
-        audience=os.environ.get("IDP_AUDIENCE", None),
-        scopes=os.environ.get("IDP_SCOPES", None),
```
Comment (gregertw) [inline cognite_toolkit/cdf_tk/bootstrap.py:59]: Any particular reason for removing the explicitness of setting the var to None? It's default, but the less you need to remember about a specific function, the better IMO

**Review by gregertw — APPROVED** [review]
LGTM

**Pull request merged**